### PR TITLE
feat: add KeyPair-to-JWK conversion

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

This PR adds a conversion method from `KeyPair` (JDK) to `JWK` (Nimbus lib) for easy use downstream.

## Why it does that

In some cases, such as when signing VerifiablePresentations, we may need the actual JWK representation, and Nimbus doesn't have 
good support for converting.

## Further notes

- the JDK does not have good support for Elliptic Curve multiplication, which we need to regenerate the PublicKey from a PrivateKey. BouncyCastle does, so I added it 
as implementation dependency. 
- some dependencies of the `identity-trust-service` module got cleaned up
- renamed the `JwsSignerVerifierFactory` to `CryptoConverter`

## Open questions

- ~~Do we want to rename the `JwsSignerVerifierFactory`? The name seems kinda weird, but I can't think of a better one.~~ -> now named `CryptConverter`
  [edit]: ~~maybe `NimbusHelper`?~~
- ~~Is the `jwt-spi` package the best place for it? probably some crypto lib module would be more suitable.~~ moved to `crypto-core`

## Linked Issue(s)

Contributes to #3743

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/.github/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/.github/blob/main/contributing/pr_etiquette.md)._

### CODE REVIEW

File: core/common/token-core/src/main/java/org/eclipse/edc/token/JwtGenerationService.java [hunk 1]
```diff
@@ -37,11 +37,11 @@
 import java.util.function.Supplier;
 
 public class JwtGenerationService implements TokenGenerationService {
-    private final JwsSignerVerifierFactory factory;
+    private final CryptoConverter factory;
```
Comment (ndr-brt) [inline core/common/token-core/src/main/java/org/eclipse/edc/token/JwtGenerationService.java:40]:
```suggestion
    private final CryptoConverter converter;
```

File: core/common/token-core/build.gradle.kts [hunk 2]
```diff
@@ -18,8 +18,10 @@ plugins {
 }
 
 dependencies {
+    api(project(":spi:common:token-spi"))
     api(project(":spi:common:jwt-spi"))
 
+    implementation(project(":extensions:common:crypto:crypto-core")) // for the CryptoConverter
```
Comment (ndr-brt) [inline core/common/token-core/build.gradle.kts:24]: given that `crypto-core` is not an extension but a library, it could stay in the `core` folder

File: spi/common/oauth2-spi/build.gradle.kts [hunk 3]
```diff
@@ -18,6 +18,7 @@ plugins {
 
 dependencies {
     api(project(":spi:common:jwt-spi"))
+    implementation(libs.nimbus.jwt)
```
Comment (ndr-brt) [inline spi/common/oauth2-spi/build.gradle.kts:21]: it should be not necessary

File: extensions/common/crypto/crypto-core/src/main/java/org/eclipse/edc/security/token/jwt/CryptoConverter.java [hunk 4]
```diff
@@ -131,16 +178,97 @@ public JWSAlgorithm getRecommendedAlgorithm(JWSSigner signer) {
...
+        for (var i = 0; i < array.length / 2; i++) {
             var temp = array[i];
             array[i] = array[array.length - 1 - i];
             array[array.length - 1 - i] = temp;
         }
         return array;
     }
 
-    private Ed25519Verifier createOctetKeyPair(PublicKey publicKey) throws JOSEException {
+    private Ed25519Verifier createEdDsaSigner(PublicKey publicKey) throws JOSEException {
```
Comment (ndr-brt) [inline extensions/common/crypto/crypto-core/src/main/java/org/eclipse/edc/security/token/jwt/CryptoConverter.java:271]:
```suggestion
    private Ed25519Verifier createEdDsaVerifier(PublicKey publicKey) throws JOSEException {
```

File: extensions/common/crypto/crypto-core/src/main/java/org/eclipse/edc/security/token/jwt/CryptoConverter.java [hunk 5]
```diff
@@ -131,16 +178,97 @@ public JWSAlgorithm getRecommendedAlgorithm(JWSSigner signer) {
...
         var edKey = (EdECPublicKey) publicKey;
         var curveName = edKey.getParams().getName();
```
Comment (ndr-brt) [inline extensions/common/crypto/crypto-core/src/main/java/org/eclipse/edc/security/token/jwt/CryptoConverter.java:273]: not part of this PR, but I noticed that the process to get the `Curve` from the `EdECKey` (both public and private) is the same and it could be extracted in a common method

Comment (paullatzelsperger) [reply extensions/common/crypto/crypto-core/src/main/java/org/eclipse/edc/security/token/jwt/CryptoConverter.java:273]: easy fix. will do.

File: core/common/token-core/build.gradle.kts (hunk 2 above)
Comment (ndr-brt) [reply core/common/token-core/build.gradle.kts:24]: currently `core` contains also library modules (as `util`), anyway I'm ok for leaving it between the extensions as well

**Review by ndr-brt — APPROVED** [review]
